src/vellum/workflows/state/base.py

```diff
@@ -536,14 +542,16 @@ def __iter__(self) -> Iterator[Tuple[Any, Any]]:
         on an instance of this class.
         """
 
-        # If the user sets a default value on state (e.g. something = "foo"), it's not on `instance_attributes` below.
-        # So we need to include class_attributes here just in case
-        class_attributes = {key: value for key, value in self.__class__.__dict__.items() if not key.startswith("_")}
-        instance_attributes = {key: value for key, value in self.__dict__.items() if not key.startswith("__")}
+        # Use lock to create a consistent snapshot of the state
+        with self.__lock__:
```

Comment:
would love to avoid locks if we can since I'm not sure what the deadlocking implications could be if not careful.